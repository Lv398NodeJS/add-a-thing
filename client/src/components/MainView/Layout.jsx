import React, { Component } from 'react';

export default class Layout extends Component {

  render() {
    const { children } = this.props;
    return (
      <>
        {children}
      </>
    );
  }
}
