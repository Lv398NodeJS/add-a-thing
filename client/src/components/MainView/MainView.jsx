import React from 'react';
import { Container, Col, Row } from 'react-bootstrap';
import '@src/App.scss';
import CreateDashboard from './CreateDashboard/CreateDashboard';
import ListOfDashboards from './ListOfDashboards/ListOfDashboards';
import WelcomeView from '@MainView/Header/WelcomeView';
import Header from '@MainView/Header/Header';

export default class MainView extends React.Component {
  render() {
    let content;
    if (localStorage.getItem('token')) {
      content = (
        <Container className="App">
          <Row>
            <Col>
              <CreateDashboard />
              <ListOfDashboards />
            </Col>
          </Row>
        </Container>
      );
    } else {
      content = <WelcomeView />;
    }

    return (
      <>
        <Header />
        {content}
      </>
    );
  }
}