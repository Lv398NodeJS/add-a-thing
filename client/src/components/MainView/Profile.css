.add-avatar{
  width: 90px;
}

.profile-container{
  background: #76b7c0;
}