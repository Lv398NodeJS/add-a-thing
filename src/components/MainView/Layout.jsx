import React, { Component } from 'react';
import Header from '../Dashboard/Header/Header';

export default class Layout extends Component {
  constructor(props) {
    super(props);

    this.state = {
      isLoggedIn: props.isLoggedIn,
    };
  }

  render() {
    const { isLoggedIn, children } = this.props;
    return (
      <>
        <Header isLoggedIn={isLoggedIn} />
        {children}
      </>
    );
  }
}
