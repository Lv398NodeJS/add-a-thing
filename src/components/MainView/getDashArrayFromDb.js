import db from '../../fire';

export default () => {
  const newDashArray = [];
  const dashboardsRef = db.database().ref('dashboards');
  dashboardsRef.on('value', (snapshot) => {
    const dashboardsSnap = snapshot.val();
    Object.keys(dashboardsSnap).forEach(dash => (
      newDashArray.push({
        id: dash,
        name: dashboardsSnap[dash].name,
        description: dashboardsSnap[dash].description,
        key: dash,
      })));
  });
  return newDashArray;
};
