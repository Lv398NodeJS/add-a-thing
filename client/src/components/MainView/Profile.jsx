import React, {Component} from 'react';
import Header from '@MainView/Header/Header';
import { Redirect } from 'react-router-dom';
import {Button, Col, Form, Row, Container} from "react-bootstrap";
import avatar from '@assets/avatar.svg';
import './Profile.css';
import {connect} from "react-redux";
import {bindActionCreators} from "redux";
import * as loginActions from '@actions/loginationActions';

export class Profile extends Component {
	constructor(props) {
		super(props);
		this.state = {
			showEdit: true,
		};
	}

	handleEdit = () => {
		this.setState({
			showEdit: false,
		});
	};

	handleSaveProfile = () => {
		const { updateName, updateEmail, updatePhone } = this.state;
		const { userData, loginationActions: { updateProfile } } = this.props;
		const updateUser = {
			id: userData.id,
			name: updateName,
			email: updateEmail,
			phone: updatePhone,
		};
		if(updateName && updateEmail && updatePhone) {
			updateProfile(updateUser);

			this.setState({
				showEdit: true,
			});
		}
	};

	handleSave = (event) => {
		const { name, value } = event.target;
		this.setState({
			[name]: value,
		});
	};

	render() {
		const { showEdit, updateName, updateEmail, updatePhone } = this.state;
		const { userData } = this.props;
		if(!localStorage.getItem('token')){
			return <Redirect to="/" />;
		}

		const content = showEdit ? (<Row>
			<Col md={{ span: 4, offset: 4 }}>
				<h1>Profile</h1>
				<Row>
					<Col className="m-1">
						<img src={avatar} alt="avatar" className="add-avatar" />
					</Col>
				</Row>
				<Form.Label>Name</Form.Label>
					<Container className="profile-container">
						{ updateName }
					</Container>
				<Form.Label>Email</Form.Label>
					<Container className="profile-container">
						{ updateEmail }
					</Container>
				<Form.Label>Phone</Form.Label>
					<Container className="profile-container">
						{ updatePhone }
					</Container>

				<Button
				className="mt-2"
				variant="primary"
				size="md"
				type="submit"
				valur="Create New Account"
				onClick={this.handleEdit}
				>
					Update profile
				</Button>
			</Col>
		</Row>) : (<Row>
			<Col md={{ span: 4, offset: 4 }}>
				<h1>Profile</h1>
				<Row>
					<Col className="m-1">
						<img src={avatar} alt="avatar" className="add-avatar" />
						<Row className="m-1">
							<Button
							className="avatar-btn"
							variant="primary"
							size="sm"
							>
								Change avatar
							</Button>
						</Row>
					</Col>
				</Row>
				<Form.Label>Name</Form.Label>
				<Form.Control
				placeholder="Your Name"
				defaultValue={userData.name}
				name="updateName"
				value={updateName}
				onChange={this.handleSave}
				/>

				<Form.Label>Email</Form.Label>
				<Form.Control
				type="email"
				placeholder="user@mail.com"
				defaultValue={userData.email}
				name="updateEmail"
				value={updateEmail}
				onChange={this.handleSave}
				/>

				<Form.Label>Phone</Form.Label>
				<Form.Control
				type="text"
				placeholder="+380965742187"
				defaultValue={userData.phone}
				name="updatePhone"
				value={updatePhone}
				onChange={this.handleSave}
				/>
				<Button
				className="mt-2"
				variant="primary"
				size="md"
				type="submit"
				valur="Create New Account"
				onClick={this.handleSaveProfile}
				>
					Save profile
				</Button>
			</Col>
		</Row>);
		return (
		<>
			<Header />
			{content}
		</>
		);
	}
}

const mapStateToProps = state => ({
	userData: state.loginationReducer.userData,
});

const mapDispatchToProps = dispatch => ({
	loginationActions: bindActionCreators(loginActions, dispatch),
});

export default connect(
	mapStateToProps,
	mapDispatchToProps,
)(Profile);