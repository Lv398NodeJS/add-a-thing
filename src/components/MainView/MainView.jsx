/* eslint-disable guard-for-in */
/* eslint-disable no-restricted-syntax */
import React, {Component} from 'react';
import {Container} from 'react-bootstrap';
import '../../App.scss';
import CreateDashboard from './CreateDashboard/CreateDashboard';
import ListOfDashboards from './ListOfDashboards/ListOfDashboards';
import db from '../../fire';
import getDashArrayFromDb from './getDashArrayFromDb';
import NavBar from "../Dashboard/Header/Header";
import Col from "react-bootstrap/es/Col";
import Row from "react-bootstrap/es/Row";

class MainView extends Component {
  constructor(props) {
    super(props);

    this.state = {
      dashboards: [],
    };
  }

  componentDidMount() {
    const dashboardsRef = db.database().ref('dashboards');
    dashboardsRef.on('value', (snapshot) => {
      const dashboardsSnap = snapshot.val();
      this.setState({
        dashboards: getDashArrayFromDb(dashboardsSnap),
      });
    });
  }

  addDashboard = (data) => {
    this.setState(prevState => ({
      dashboards: [...prevState.dashboards, data],
    }));
  }

  deleteDashboard = (data) => {
    this.setState(prevState => ({
      dashboards: prevState.dashboards.filter(dash => dash.innerId !== data.innerId),
    }));
  }

  render() {
    const {dashboards} = this.state;
    return (
      <>
        <NavBar />
        <Container className="App">
          <Row>
            <Col>
              <CreateDashboard
                handleAddDashboard={this.addDashboard}
              />
              <ListOfDashboards
                dashboardArray={dashboards}
                deleteDashboard={this.deleteDashboard}
              />
            </Col>
          </Row>
        </Container>
      </>
    );
  }
}

export default MainView;
