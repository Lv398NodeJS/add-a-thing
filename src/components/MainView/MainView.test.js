import React from 'react';
import { shallow } from 'enzyme';
import MainView from './MainView';

describe('MyComponent', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<MainView debug />);

    expect(component).toMatchSnapshot();
  });
});
