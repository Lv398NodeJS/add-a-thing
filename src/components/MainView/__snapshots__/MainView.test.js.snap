// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyComponent should render correctly in "debug" mode 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainView
    debug={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <NavBar />,
        <ForwardRef
          as="div"
          className="App"
          fluid={false}
        >
          <ForwardRef(render)>
            <ForwardRef
              as="div"
            >
              <CreateDashboard
                addDashboard={[Function]}
              />
              <ListOfDashBoards
                dashboardArray={Array []}
                deleteDashboard={[Function]}
              />
            </ForwardRef>
          </ForwardRef(render)>
        </ForwardRef>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "as": "div",
          "children": <ForwardRef(render)>
            <ForwardRef
              as="div"
            >
              <CreateDashboard
                addDashboard={[Function]}
              />
              <ListOfDashBoards
                dashboardArray={Array []}
                deleteDashboard={[Function]}
              />
            </ForwardRef>
          </ForwardRef(render)>,
          "className": "App",
          "fluid": false,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <ForwardRef
              as="div"
            >
              <CreateDashboard
                addDashboard={[Function]}
              />
              <ListOfDashBoards
                dashboardArray={Array []}
                deleteDashboard={[Function]}
              />
            </ForwardRef>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "as": "div",
              "children": Array [
                <CreateDashboard
                  addDashboard={[Function]}
                />,
                <ListOfDashBoards
                  dashboardArray={Array []}
                  deleteDashboard={[Function]}
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "addDashboard": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "dashboardArray": Array [],
                  "deleteDashboard": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "as": "div",
              },
              "displayName": "Col",
              "render": [Function],
            },
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": undefined,
            "displayName": "Bootstrap(Row)",
            "propTypes": undefined,
            "render": [Function],
          },
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "as": "div",
            "fluid": false,
          },
          "displayName": "Container",
          "render": [Function],
        },
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <NavBar />,
          <ForwardRef
            as="div"
            className="App"
            fluid={false}
          >
            <ForwardRef(render)>
              <ForwardRef
                as="div"
              >
                <CreateDashboard
                  addDashboard={[Function]}
                />
                <ListOfDashBoards
                  dashboardArray={Array []}
                  deleteDashboard={[Function]}
                />
              </ForwardRef>
            </ForwardRef(render)>
          </ForwardRef>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "div",
            "children": <ForwardRef(render)>
              <ForwardRef
                as="div"
              >
                <CreateDashboard
                  addDashboard={[Function]}
                />
                <ListOfDashBoards
                  dashboardArray={Array []}
                  deleteDashboard={[Function]}
                />
              </ForwardRef>
            </ForwardRef(render)>,
            "className": "App",
            "fluid": false,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <ForwardRef
                as="div"
              >
                <CreateDashboard
                  addDashboard={[Function]}
                />
                <ListOfDashBoards
                  dashboardArray={Array []}
                  deleteDashboard={[Function]}
                />
              </ForwardRef>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "as": "div",
                "children": Array [
                  <CreateDashboard
                    addDashboard={[Function]}
                  />,
                  <ListOfDashBoards
                    dashboardArray={Array []}
                    deleteDashboard={[Function]}
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "addDashboard": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "dashboardArray": Array [],
                    "deleteDashboard": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": Object {
                  "as": "div",
                },
                "displayName": "Col",
                "render": [Function],
              },
            },
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": undefined,
              "displayName": "Bootstrap(Row)",
              "propTypes": undefined,
              "render": [Function],
            },
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "as": "div",
              "fluid": false,
            },
            "displayName": "Container",
            "render": [Function],
          },
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;

exports[`MyComponent should render correctly with no props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <MainView />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <NavBar />,
        <ForwardRef
          as="div"
          className="App"
          fluid={false}
        >
          <ForwardRef(render)>
            <ForwardRef
              as="div"
            >
              <CreateDashboard
                addDashboard={[Function]}
              />
              <ListOfDashBoards
                dashboardArray={Array []}
                deleteDashboard={[Function]}
              />
            </ForwardRef>
          </ForwardRef(render)>
        </ForwardRef>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {},
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "as": "div",
          "children": <ForwardRef(render)>
            <ForwardRef
              as="div"
            >
              <CreateDashboard
                addDashboard={[Function]}
              />
              <ListOfDashBoards
                dashboardArray={Array []}
                deleteDashboard={[Function]}
              />
            </ForwardRef>
          </ForwardRef(render)>,
          "className": "App",
          "fluid": false,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": <ForwardRef
              as="div"
            >
              <CreateDashboard
                addDashboard={[Function]}
              />
              <ListOfDashBoards
                dashboardArray={Array []}
                deleteDashboard={[Function]}
              />
            </ForwardRef>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "as": "div",
              "children": Array [
                <CreateDashboard
                  addDashboard={[Function]}
                />,
                <ListOfDashBoards
                  dashboardArray={Array []}
                  deleteDashboard={[Function]}
                />,
              ],
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "addDashboard": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "class",
                "props": Object {
                  "dashboardArray": Array [],
                  "deleteDashboard": [Function],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": Object {
                "as": "div",
              },
              "displayName": "Col",
              "render": [Function],
            },
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": undefined,
            "displayName": "Bootstrap(Row)",
            "propTypes": undefined,
            "render": [Function],
          },
        },
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "defaultProps": Object {
            "as": "div",
            "fluid": false,
          },
          "displayName": "Container",
          "render": [Function],
        },
      },
    ],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <NavBar />,
          <ForwardRef
            as="div"
            className="App"
            fluid={false}
          >
            <ForwardRef(render)>
              <ForwardRef
                as="div"
              >
                <CreateDashboard
                  addDashboard={[Function]}
                />
                <ListOfDashBoards
                  dashboardArray={Array []}
                  deleteDashboard={[Function]}
                />
              </ForwardRef>
            </ForwardRef(render)>
          </ForwardRef>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "as": "div",
            "children": <ForwardRef(render)>
              <ForwardRef
                as="div"
              >
                <CreateDashboard
                  addDashboard={[Function]}
                />
                <ListOfDashBoards
                  dashboardArray={Array []}
                  deleteDashboard={[Function]}
                />
              </ForwardRef>
            </ForwardRef(render)>,
            "className": "App",
            "fluid": false,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "children": <ForwardRef
                as="div"
              >
                <CreateDashboard
                  addDashboard={[Function]}
                />
                <ListOfDashBoards
                  dashboardArray={Array []}
                  deleteDashboard={[Function]}
                />
              </ForwardRef>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "as": "div",
                "children": Array [
                  <CreateDashboard
                    addDashboard={[Function]}
                  />,
                  <ListOfDashBoards
                    dashboardArray={Array []}
                    deleteDashboard={[Function]}
                  />,
                ],
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "addDashboard": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "class",
                  "props": Object {
                    "dashboardArray": Array [],
                    "deleteDashboard": [Function],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
              ],
              "type": Object {
                "$$typeof": Symbol(react.forward_ref),
                "defaultProps": Object {
                  "as": "div",
                },
                "displayName": "Col",
                "render": [Function],
              },
            },
            "type": Object {
              "$$typeof": Symbol(react.forward_ref),
              "defaultProps": undefined,
              "displayName": "Bootstrap(Row)",
              "propTypes": undefined,
              "render": [Function],
            },
          },
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "defaultProps": Object {
              "as": "div",
              "fluid": false,
            },
            "displayName": "Container",
            "render": [Function],
          },
        },
      ],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
